pkgs/applications/networking/browsers/chromium-git/default.nix

```diff
@@ -0,0 +1,197 @@
+      ];
+
+      postPatch = ''
+        ( cd src
+          # We want to be able to specify where the sandbox is via CHROME_DEVEL_SANDBOX
+          substituteInPlace sandbox/linux/suid/client/setuid_sandbox_host.cc \
+            --replace \
+              'return sandbox_binary;' \
+              'return base::FilePath(GetDevelSandboxPath());'
+
+          substituteInPlace services/audio/audio_sandbox_hook_linux.cc \
+            --replace \
+              '/usr/share/alsa/' \
+              '${alsaLib}/share/alsa/' \
+            --replace \
+              '/usr/lib/x86_64-linux-gnu/gconv/' \
+              '${glibc}/lib/gconv/' \
+            --replace \
+              '/usr/share/locale/' \
+              '${glibc}/share/locale/'
+
+          sed -i -e 's@"\(#!\)\?.*xdg-@"\1${xdg_utils}/bin/xdg-@' \
+            chrome/browser/shell_integration_linux.cc
+
+          sed -i -e '/lib_loader.*Load/s!"\(libudev\.so\)!"${systemd.lib}/lib/\1!' \
+            device/udev_linux/udev?_loader.cc
+
+          sed -i -e '/libpci_loader.*Load/s!"\(libpci\.so\)!"${pciutils}/lib/\1!' \
+            gpu/config/gpu_info_collector_linux.cc
+
+          # Allow to put extensions into the system-path.
+          sed -i -e 's,/usr,/run/current-system/sw,' chrome/common/chrome_paths.cc
+
+          ${lib.optionalString stdenv.isAarch64 ''
+              substituteInPlace build/toolchain/linux/BUILD.gn \
+                --replace 'toolprefix = "aarch64-linux-gnu-"' 'toolprefix = ""'
+           ''}
+
+          patchShebangs --build .
+
+          mkdir -p third_party/node/linux/node-linux-x64/bin
+          ln -s ${nodejs}/bin/node                    third_party/node/linux/node-linux-x64/bin/node      || true
+
+          mkdir -p third_party/llvm-build/Release+Asserts/bin
+          ln -s ${llvmPackages_8.clang}/bin/clang     third_party/llvm-build/Release+Asserts/bin/clang    || true
+          ln -s ${llvmPackages_8.clang}/bin/clang++   third_party/llvm-build/Release+Asserts/bin/clang++  || true
+          ln -s ${llvmPackages_8.llvm}/bin/llvm-ar    third_party/llvm-build/Release+Asserts/bin/llvm-ar  || true
+
+          echo 'build_with_chromium = true'                > build/config/gclient_args.gni
+          echo 'checkout_android = false'                 >> build/config/gclient_args.gni
+          echo 'checkout_android_native_support = false'  >> build/config/gclient_args.gni
+          echo 'checkout_nacl = false'                    >> build/config/gclient_args.gni
+          echo 'checkout_oculus_sdk = false'              >> build/config/gclient_args.gni
+        )
+      '';
+
+      configurePhase = ''
+        ( cd src
+          gn gen ${lib.escapeShellArg "--args=${gnToString gnFlags}"} out/Release
+        )
+      '';
+
+      buildPhase = ''
+        ( cd src
+          ninja -C out/Release chrome
+
+          find chrome/test/chromedriver -exec touch -t 198001010000.00 {} +   # fix zip issues with timestamps older than 1980
```

Comment:
Could we use hooks here? `ensureNewerSourcesForZipFilesHook`

Reply:
It might make difficult further refactoring: `chromedriver` shouldn't be here at all, it must be a separate derivation or output. I build it here together with `chromium` to reduce the build time and, more importantely, to help `chromedriver` to fild proper `chromium`, so `selenium` could work with many versions.
But it is a separate project and should eventually have its own derivation.

Reply:
I removed `chromedriver` from this derivation